test/cg.jl

```diff
@@ -1,45 +1,44 @@
 using IterativeSolvers
-using FactCheck
 using LinearMaps
+using Base.Test
 
 srand(1234321)
-
 include("poisson_matrix.jl")
 
-facts("cg") do
+@testset "Conjugate Gradients" begin
 
-context("Small full system") do
-    N = 10
-    A = randn(N, N)
-    A = A' * A
-    rhs = randn(N)
-    tol = 1e-12
+@testset "Small full system" begin
+    n = 10
 
-    x = cg(A, rhs; tol=tol, maxiter=2N)
-    @fact norm(A*x - rhs) --> less_than(cond(A) * √tol)
+    @testset "Matrix{$T}" for T in (Float32, Float64, Complex64, Complex128)
+        A = rand(T, n, n)
+        A = A' * A + I
+        b = rand(T, n)
+        tol = √eps(real(T))
 
-    x,ch = cg(A, rhs; tol=tol, maxiter=2N, log=true)
-    @fact norm(A * x - rhs) --> less_than(cond(A) * √tol)
-    @fact ch.isconverged --> true
+        x,ch = cg(A, b; tol=tol, maxiter=2n, log=true)
+        @test norm(A*x - b) / norm(b) ≤ tol
+        @test ch.isconverged
 
-    # If you start from the exact solution, you should converge immediately
-    x2, ch2 = cg!(A \ rhs, A, rhs; tol=10tol, log=true)
-    @fact niters(ch2) --> less_than_or_equal(1)
-    @fact nprods(ch2) --> less_than_or_equal(2)
+        # If you start from the exact solution, you should converge immediately
+        x,ch = cg!(A \ b, A, b; tol=10tol, log=true)
+        @test niters(ch) ≤ 1
+        @test nprods(ch) ≤ 2
 
-    # Test with cholfact should converge immediately
-    F = cholfact(A)
-    x2, ch2 = cg(A, rhs; Pl=F, log=true)
-    @fact niters(ch2) --> less_than_or_equal(2)
-    @fact nprods(ch2) --> less_than_or_equal(2)
+        # Test with cholfact should converge immediately
+        F = cholfact(A)
+        x,ch = cg(A, b; Pl=F, log=true)
+        @test niters(ch) ≤ 2
+        @test nprods(ch) ≤ 2
 
-    # All-zeros rhs should give all-zeros lhs
-    x0 = cg(A, zeros(N))
-    @fact x0 --> zeros(N)
+        # All-zeros rhs should give all-zeros lhs
+        x0 = cg(A, zeros(T, n))
+        @test x0 == zeros(T, n)
+    end
 end
 
-context("Sparse Laplacian") do
-    A = poisson_matrix(Float64, 32, 3)
+@testset "Sparse Laplacian" begin
+    A = poisson_matrix(Float64, 10, 3)
```

Comment:
At the moment is generates a matrix of size 32^3 = 32.768, which is rather large for a test.